src/lib/reference.ts

```diff
@@ -0,0 +1,39 @@
+import type { Blueprint } from '@seamapi/blueprint'
+import type Metalsmith from 'metalsmith'
+
+const sdks = ['javascript']
+
+export const reference = (
+  files: Metalsmith.Files,
+  metalsmith: Metalsmith,
+): void => {
+  const metadata = metalsmith.metadata()
+
+  for (const route of (metadata as Blueprint).routes ?? []) {
+    for (const endpoint of route.endpoints) {
+      const k = `api${endpoint.path}.md`
+      files[k] = {
+        contents: Buffer.from('\n'),
+      }
+      const file = files[k] as Partial<TemplateContext>
+      file.layout = 'api-reference.hbs'
+      file.endpoint = endpoint
```

Comment:
@mastafit We can add something like this instead if we want build a more complicated template context
```suggestion
      setFileContext(file)
```

Reply:
@razor-x Sure, just give me some requirements for this one, what exactly such function should do?

Reply:
It depends on what context we need for the template. So if we write the handlebars template and only need `endpoint`, then we are done. But we may need to extend the context, add to it, or transform it, if `endpoint` is not enough or hard to use in the template.

Ideally, blueprint would already have what we need so `element` should be a good context for the template, but at this stage we are still learning what we need in context.

If this gets too complex, we can move logic into handlebars helpers or back into the blueprint. 

For example, one thing we want to do is update URLs to be relative in GitBook, e.g., if a description contains a Markdown link to to `https://docs.seam.co/latest/core-concepts/overview` we really want this to link to `../core-concepts/overview` or whatever the correct internal GitBook linking should be.

Another example is we want to generate links to other resources, e.g., `The `acs_system` is...` should might get transformed to `The [`acs_system`](../acs_system)` is...`.

We can implement those two things later, I'm just using them as an example. But either it would be handled inside `setFileContext` or in the template via a helper like `{{ element.description | linkResources | relativeLinks }}`.

This PR does not include registering handlebars helpers, but it should be simple to add that under `src/lib`.